runtime/src/main/cpp/Memory.h

```diff
@@ -329,6 +329,12 @@ struct MetaObjHeader {
     // Strong reference to the counter object.
     ObjHeader* counter_;
   } WeakReference;
+  struct {
+    // Leak detector's previous list element.
+    ObjHeader* previous_;
+    // Leak detector's next list element.
+    ObjHeader* next_;
+  } LeakDetector;
```

Comment:
This is kinda space-inefficient. Can we avoid this?

Reply:
Replaced it with a global hashmap and a list, which are only allocated if someone asks for this leak checker (this isn't currently the case, will fix).